californium-core/src/main/java/org/eclipse/californium/core/network/stack/BlockwiseLayer.java

```diff
@@ -630,6 +630,12 @@ public void receiveResponse(final Exchange exchange, final Response response) {
 				case REQUEST_ENTITY_INCOMPLETE: // 4.08
 					// we seem to have uploaded blocks not in expected order
 				case REQUEST_ENTITY_TOO_LARGE: // 4.13
+					if (response.getOptions().hasBlock1() && exchange.getCurrentRequest().getOptions().hasBlock1()
+							&& response.getOptions().getBlock1().getSzx() < exchange.getCurrentRequest().getOptions()
+									.getBlock1().getSzx()) {
+						handleBlock1Response(exchange, response);
+						return;
+					}
```

Comment:
I'm not sure, if a transfer should change the blocksize in the middle of the transfer. I would assume, this is only done at the begin.

Reply:
I understand your point but some part of the RFC makes me thinks that this could maybe be allowed. As this was not really more costly to handle it  I decided to go for it.

If you insist I can only support it for first answer.

Reply:
[block1 size negotiation](https://tools.ietf.org/html/rfc7959#page-24)

I'm not sure, where changing the blocksize in the middle should be mentioned in RFC 7959.
Could you provide a link to that?

Reply:
Not clearly mention, I just see some sentences make me think that maybe this is allowed:  

> The present specification allows the server to return this response
   code at any time during a Block1 transfer to indicate that it does
   not currently have the resources to store blocks for a transfer that
   it would intend to implement in an atomic fashion.

(but reading it again, this is maybe more about stopping the transfer than negotiating the block size)

>    Finally, a 4.13 response to a request with a Block1 Option (control
   usage, see Section 2.3) where the response carries a smaller SZX in
   its Block1 Option is a hint to try that smaller SZX.

(not really explicit say this is only for first transfer) 

> Finally, the SZX block size given in a control Block1 Option
         indicates the largest block size preferred by the server for
         transfers toward the resource that is the same or smaller than
         the one used in the initial exchange; the client SHOULD use
         this block size or a smaller one in all further requests in the
         transfer sequence, even if that means changing the block size
         (and possibly scaling the block number accordingly) from now
         on.

(this makes maybe thing that this is only on first exchange)

Reply:
A question in the IETF core mailing list may bring clarification.

Reply:
That has now grown to a larger couple o f lines. Please consider to move that into a own method.